### PR TITLE
Generic NMEA driver

### PR DESCRIPTION
**Pull request by bkueng**
From https://github.com/PX4/PX4-GPSDrivers/pull/74 with some cleanup & fixes.

Tested with a u-blox m8n set to NMEA output.

### CODE REVIEW

File: src/nmea.cpp [hunk 1]
```diff
@@ -0,0 +1,1015 @@
...
+	int iRet = 0;
+
+	switch (_decode_state) {
+	/* First, look for sync1 */
+	case NMEADecodeState::uninit:
+		if (b == '$') {
+			_decode_state = NMEADecodeState::got_sync1;
+			_rx_buffer_bytes = 0;
+			_rx_buffer[_rx_buffer_bytes++] = b;
+		}
```
Comment (lukegluke) [inline src/nmea.cpp:916]:
```
} else if (b == RTCM3_PREAMBLE && _rtcm_parsing) {
	_decode_state = NMEADecodeState::decode_rtcm3;
	_rtcm_parsing->addByte(b);

}
```

@bkueng, probably `_rtcm_parsing` should be added same as in ashtech driver for PPK (0acaf6185245382389f7f6019cee6537d8f5a8e6) .

**Comment by Kim-zhi-jiang** [comment]
@bkueng Positioned at first, move then circle around(in master version of yesterday)

UWB transmission to info mode(only 3):
$GPGGA,000522.953,2231.08747980,N,11354.04484670,E,1,04,0.030,-0.681,M,0.000,M,,0000*43
$GPGSA,A,3,0,1,2,3,,,,,,,,,2.550,0.030,2.549*39
$GPRMC,000522.953,A,2231.08747980,N,11354.04484670,E,000.007,125.498,080318,,,A*62

log: [here](https://logs.px4.io/plot_app?log=c825a49f-cc93-4f27-9515-de408631f425)

**Comment by bkueng** [comment]
Thanks for testing @Kim-zhi-jiang, which exact GPS is that?

File: src/nmea.cpp (hunk 1 above)
Comment (bkueng) [reply src/nmea.cpp:916]: Yes, can you send a PR?

**Comment by Kim-zhi-jiang** [comment]
@bkueng [nooploop](https://www.nooploop.com/)
nnn

File: src/nmea.cpp (hunk 1 above)
Comment (lukegluke) [reply src/nmea.cpp:916]: Done: https://github.com/PX4/PX4-GPSDrivers/pull/88 (but haven't tested it)

**Pull request merged**